sdk/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -119,6 +124,23 @@ public partial class Deployment
             var resourceMonitorSupportsAliasSpecs = await MonitorSupportsAliasSpecs().ConfigureAwait(false);
             var aliases = await PrepareAliases(res, options, resourceMonitorSupportsAliasSpecs).ConfigureAwait(false);
 
+            var transforms = new List<Pulumirpc.Callback>();
+            if (options.XResourceTransforms.Count > 0)
+            {
+                var resourceMonitorSupportsTransforms = await MonitorSupportsTransforms().ConfigureAwait(false);
+                if (!resourceMonitorSupportsTransforms)
+                {
+                    throw new Exception("Engine does not support resource transformations");
```

Comment:
Should this be something like `InvalidOperationException` and have a nicer message like in the other SDKs, e.g. "upgrade your CLI"?

Reply:
Updated to match what we do for the DeletedWith option.